python/dgl/dataloading/dataloader.py

```diff
@@ -72,7 +72,7 @@ def __next__(self):
         # convert the type-ID pairs to dictionary
         type_ids = batch[:, 0]
         indices = batch[:, 1]
-        type_ids_sortidx = torch.argsort(type_ids)
+        _, type_ids_sortidx = torch.sort(type_ids, stable=True)
```

Comment:
I guess it's better to raise a warning for older versions of PyTorch when stable sorting is not available and resort to non-stable version (or does PyTorch do that already?)

Reply:
I agree.
It seems older PyTorch will not resort to non-stable version, it just raises an invalid arguments error.
I add the torch version check and raise a warning for older versions of Pytorch. One problem is the warning will also be raised when `shuffle` is `True`, the disorder of the output_nodes should not matter in this scenario.

Reply:
> One problem is the warning will also be raised when `shuffle` is `True`, the disorder of the output_nodes should not matter in this scenario.

Then you could choose to only raise the warning if `shuffle` is False.

> by the way, dgl_warning() seems will repeat the first line?

You mean the DGLWarning will be repeated for each process?  I guess there's not a good way to handle that, but the purpose of warnings is to let users know that there is a problem when `shuffle` is False.
Another solution would be using `np.argsort` (that has a stable option) rather than `torch.sort` if `type_ids` is on CPU, and just raise an error if `type_ids` is on GPU, PyTorch version is too old, and `shuffle=False`.

Reply:
> Then you could choose to only raise the warning if shuffle is False.

I thought about this solution, but there is no shuffle parameter passed in _TensorizedDatasetIter(object).

> You mean the DGLWarning will be repeated for each process?

what I mean here is the first line of the message in dgl_warning will be printed again. For example, below is the full output, the line `The current output_nodes are out of order even if set shuffle to False in Dataloader ` will be printed again at the end of the message.

>DGLWarning: The current output_nodes are out of order even if set shuffle to False in Dataloader, the reason is that the current version of torch dose not support stable sort. Please update torch to 1.10.0 or higher to fix it.
' The current output_nodes are out of order even if set shuffle to False in Dataloader, '
set `line=''` in `_default_formatwarning(message, category, filename, lineno, line=None)`  may avoid this.
```
def dgl_warning_format(message, category, filename, lineno, line=None):
    """Format DGL warnings."""
    if isinstance(category, DGLWarning):
        return "DGL Warning: {}\n".format(message)
    else:
        return _default_formatwarning(message, category, filename, lineno, line=None)
```

> Another solution would be using np.argsort (that has a stable option) rather than torch.sort if type_ids is on CPU, and just raise an error if type_ids is on GPU, PyTorch version is too old, and shuffle=False.

What I was thinking about before was trying to get the same behavior for the cpu or gpu,  and avoid adding too much code.

Reply:
Decided to pass in a shuffle argument to the iterator.

I'm also thinking of the options to just deprecating PyTorch <1.10.  @jermainewang you mentioned previously that we only plan to support the last three stable versions.  Did we formally announce this policy?
(Not an actual review)